src/loggers.jl

```diff
@@ -473,3 +474,34 @@ function Base.show(io::IO, aol::AverageObservableLogger)
             aol.current_block_size * length(aol.block_averages),
             " samples collected for observable ", aol.observable)
 end
+
+mutable struct ReplicaExchangeLogger
+    n_replicas::Int
+    n_exchanges::Int
+    indices::Vector{Tuple{Int,Int}}
+    steps::Vector{Int}
+    delta::Vector{Float64}
+    end_step::Int
+end
+
+ReplicaExchangeLogger(n_replicas::Int) = ReplicaExchangeLogger(n_replicas, 0, Tuple{Int, Int}[], Int[], Float64[], 0)
+
+function log_property!(
+                    rexl::ReplicaExchangeLogger,
+                    sys::ReplicaSystem,
+                    neighbors=nothing,
+                    step_n::Int=0;
+                    indices::Tuple{Int,Int},
+                    delta::Float64,
+                    n_threads=Threads.nthreads())
+    push!(rexl.indices, indices)
+    push!(rexl.steps, step_n+rexl.end_step)
+    push!(rexl.delta, delta)
+    rexl.n_exchanges += 1
+end
+
+function finish_logs!(rexl::ReplicaExchangeLogger)
```

Comment:
I think this should record the final step number regardless of recorded exchanges. Perhaps pass `n_steps` to this function and add that to the current `end_step`?